### PR TITLE
fix(core): align resource_identifier namespace with OIDC provider

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Issue:

I use to mix use of resource identifiers and resource indicators to represent the same concept.  Auth0 uses the `resource identifier` in their web console to represent the API name of a resource.   However among the OIDC Provider's code and doc. Resource indicator is used widely.  Thus for better convenience:

Replace `resource_identifier` name with `resource_indicator` both in the table schema and code 
Align the usage with OIDC Provider

## Linear Issue Reference

LOG-1430

## Testing

UT passed
TODO: will update the dev DB and restart Logto dev server once this PR is approved

@logto-io/eng

### CODE REVIEW

File: packages/schemas/src/db-entries/resource.ts [hunk 1]
```diff
@@ -7,21 +7,21 @@ import { GeneratedSchema, Guard } from '../foundations';
...
 export type Resource = {
   id: string;
   name: string;
-  identifier: string;
+  indicator: string;
   accessTokenTtl: number;
 };
 
 const createGuard: Guard<CreateResource> = z.object({
   id: z.string(),
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/schemas/src/db-entries/resource.ts:22]: Let's discuss it offline tomorrow. Did have a separate issue tracking this. One situation I'm concerning is, do we need a resource detail page. If so the query would be 'get resource by indicator'?  The URL will look like logto.admin/resources/${encoded resource indicator}. not sure if it is user-friendly.   Any way technically id is not required.

File: packages/schemas/tables/resources.sql [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
-    identifier text not null unique, /* resource indicator also used as audience */
+    indicator text not null unique, /* resource indicator also used as audience */
     access_token_ttl bigint not null default(86400), /* expiration value in seconds, default is 24h */
     primary key (id)
+);
+
+create unique index resources__indicator
+on resources (
+    indicator
 );
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/schemas/tables/resources.sql]: added, having a lint issue track this.

File: packages/schemas/src/db-entries/resource.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (gao-sun) [reply packages/schemas/src/db-entries/resource.ts:22]: i see, let's discuss it tmw

[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/schemas/src/db-entries/resource.ts:22]: or in our core API, `id` is useful since indicator may contain url-unfriendly chars.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**